internal/driver/nodeserver_helpers.go

```diff
@@ -114,6 +91,24 @@ func validateNodeExpandVolumeRequest(req *csi.NodeExpandVolumeRequest) error {
 	return nil
 }
 
+// validateNodePublishVolumeRequest validates the node publish volume request.
+// It checks the volume ID, staging target path, target path, and volume capability in the provided request.
+func validateNodePublishVolumeRequest(req *csi.NodePublishVolumeRequest) error {
+	if req.GetVolumeId() == "" {
+		return errNoVolumeID
+	}
+	if req.GetStagingTargetPath() == "" {
+		return errNoStagingTargetPath
+	} 
+	if req.GetTargetPath() == "" {
+		return errNoTargetPath
+	}
+	if req.GetVolumeCapability() == nil {
+		return errNoVolumeCapability
+	}
+	return nil
+}
+
```

Comment:
`validateNodeUnpublishVolumeRequest` is a subset of the validation performed here. Can we remove the unpublish function and pass an operation to this function instead?
```suggestion
// validateNodePublishVolumeRequest validates the node publish volume request.
// It checks the volume ID, staging target path, target path, and volume capability in the provided request.
func validateNodePublishVolumeRequest(req *csi.NodePublishVolumeRequest) error {
	if req.GetVolumeId() == "" {
		return errNoVolumeID
	}
	if req.GetStagingTargetPath() == "" {
		return errNoStagingTargetPath
	} 
	if req.GetTargetPath() == "" {
		return errNoTargetPath
	}
	if req.GetVolumeCapability() == nil {
		return errNoVolumeCapability
	}
	return nil
}

```
```
// validateNodePublishVolumeRequest validates the node publish volume request.
// It checks the volume ID and target path for all operations, as well as the staging target path and volume capability ifirn the provided request.
func validateNodePublishVolumeRequest(req *csi.NodePublishVolumeRequest, requestType string) error {
	if req.GetVolumeId() == "" {
		return errNoVolumeID
	}
	if req.GetTargetPath() == "" {
		return errNoTargetPath
	}
	if requestType == "publish" {
		if req.GetStagingTargetPath() == "" {
			return errNoStagingTargetPath
		} 
		if req.GetVolumeCapability() == nil {
			return errNoVolumeCapability
		}
	}
	return nil
}

```

Reply:
I can try and combine all of the validation funcs in a following PR to limit the scope